File: ArrayFuntion.cpp
#include <iostream>

using namespace std ; 

class Solution {
    
    public : 
    int removeDulicates( int num[] ) {          
        num[0] = 1 ;                        // Tham chieu mang    qua     ham      ntn 
        num[1] = 2 ; 
        num[2] = 3 ; 
        num[3] = 4 ; 
        num[4] = 5 ; 
        num[5] = 6 ; 
        num[6] = 7 ; 
        return 7; 
    }

}; 

int main(){
 
    // Chua thay doi nums 
    int nums[9] =  { 3, 4, 7 ,1 ,9 , 10 , 3 , 10 , 30 } ; 
    //  int sizeNums = nums.size() ;        mang trong C++   ko the    DEM = size() or length 
    cout<<"nums old: " ; 
    for( int i = 0 ; i <  9 ; i++ ){
        cout<< nums[i] << " " ; 
    }
    cout<< endl ; 


    // Da thay doi nums[]   = funtion + array ; 
    Solution a; 
    int b = a.removeDulicates( nums ) ;      // Tham chieu mang    qua     ham      ntn 
    cout << "SL OOP Funtion nums: " <<  b << endl ; 
    cout<< "nums new: " ; 
    for( int i = 0 ; i <  b ; i++ ){
        cout<< nums[i] << " " ;           
    }


    return 0; 
}

File: ArrayFuntion.java
public class ArrayFuntion {
    public int removeDulicates( int[] nums ){

        int sizeNums = nums.length ;
        int k = 0 ;
        nums[k] = nums[0] ;
        for( int i = 0 ; i < sizeNums ; i++ ){
            if( nums[k] != nums[i]  ) {
                k++;
                nums[k] = nums[i];
            }
        }
        return k + 1 ;

        /*
        int i=0;
        int count =1;
        for(int j=1;j<nums.length;j++){
            if(nums[i] == nums[j]) 
                continue;
            else
                nums[++i] = nums[j];
        }
        return i+1;
        */
    }

    public static void main(String[] args) {
        int[] nums = { 0,0,1,1,1 , 7 , 2 , 2 ,3 , 3 } ;
        ArrayFuntion aB = new ArrayFuntion() ;
        int sizeA = aB.removeDulicates( nums )  ;
        System.out.println(  sizeA   ) ;
        System.out.println( sizeA );
        for( int i = 0 ;  i < sizeA ; i++ ){
            System.out.print( nums[i] + " " );
        }
    }

}